src/coreclr/vm/class.cpp

```diff
@@ -1502,7 +1502,7 @@ int MethodTable::GetVectorSize()
             // We need to verify that T (the element or "base" type) is a primitive type.
             TypeHandle typeArg = GetInstantiation()[0];
             CorElementType corType = typeArg.GetSignatureCorElementType();
-            if (corType >= ELEMENT_TYPE_I1 && corType <= ELEMENT_TYPE_R8)
+            if (((corType >= ELEMENT_TYPE_I1) && (corType <= ELEMENT_TYPE_R8)) || (corType == ELEMENT_TYPE_I) || (corType == ELEMENT_TYPE_U))
```

Comment:
The VM was not correctly handling `nint` or `nuint` for the purposes of `GetVectorSize()` which was impacting downstream decisions about whether it was HFA/HVA and how it should be handled.